docs/testing.md

```diff
@@ -0,0 +1,32 @@
+# Testing
+
+When writing tests for new functionality there are some things to be
+aware of.
+
+Because osbs-client is used to convey queries to and responses from a
+server, unit testing needs to be able to provide the server-side part
+of that exchange.
+
+This is achieved in tests/fake_api.py, which provides a pytest fixture
+called "openshift". When unit tests call for an argument named
+"openshift", the fixture provides this value by creating an instance
+of Openshift but using a special class, Connection, to provide the HTTP
+functionality.
+
+Other fixtures, such as "osbs" and "osbs106", make use of this fixture
+as well.
+
+For all these fixtures, HTTP requests made during the unit test are
+intercepted and handled by the Connection class, which maps requests
+and methods to files within the tests/mock_jsons directory.
+
+When new functionality is going to make a request not already handled
+by this class, you will need to:
+
+* capture the response returned by the server
```

Comment:
We should recommend a way of doing that - IIRC a curl request would be sufficient